content/blog/2021/10/new-architecture-to-ease-kubewarden-administrators-lives.md

```diff
@@ -0,0 +1,257 @@
+---
+title: A new architecture to ease Kubewarden administrators' lives
+authors:
+    - Víctor Cuadrado Juan
+date: 2021-10-01
+---
+
+We are pleased to announce a new architecture for the Kubewarden stack, in line
+with its journey to maturity:
+
+The introduction of a **PolicyServer** Custom Resource Definition which allows
+users to describe a policy-server Deployment, together with binding
+**ClusterAdmissionPolicies** to a specific **PolicyServer** instance.
+
+These 2 changes are accompanied by a multitude of improvements to make Kubewarden
+more comfortable for Kubernetes Administrators, such as validation for
+Kuberwarden Custom Resources, improvements in Helm Charts, and Status and
+Conditions for **ClusterAdmissionPolicies**.
+
+For a comprehensive list, see the [release notes](TODO) of our latest Helm
+charts.
+
+## How does it look like?
+
+In previous versions, the Kubewarden Controller instantiated a single Deployment
+of policy-server. That policy-server was configured via a ConfigMap, which
+contained the Deployment options (image, replicas, …), and a list of policies to
+load compiled from the instantiated ClusterAdmissionPolicies, with information
+on where to pull them from, their configuration options and so on.
+
+With the addition of the new **PolicyServer** Custom Resource, administrators
+have a better UX (just define the Custom Resources), can define more than 1
+policy-server Deployment, and get to select what ClusterAdmissionPolicies target
+which PolicyServers. Let's see a diagram of the new architecture:
+
+{{< figure class="center" src="/images/how-it-works-kubewarden.svg" width="90%" alt="new architecture diagram">}}
+
+On the diagram, notice the 2 separate PolicyServer Deployments in cyan and mauve
+(right), created as specified in the 2 PolicyServer resources (left).
+
+Each policy server loads different policies, specified in the
+ClusterAdmissionPolicies. The new PolicyServer Custom Resource is cluster-wide,
+which means that is identifiable by its unique `name`. Here is an example of a
+PolicyServer named `tenant-a`:
+
+```yaml
+---
+apiVersion: policies.kubewarden.io/v1alpha2
+kind: PolicyServer
+metadata:
+  name: tenant-a
+spec:
+  image: ghcr.io/kubewarden/policy-server:latest
+  replicas: 1
+```
```

Comment:
Do you think it would be good to mention that every policy server accepts a service account too, that will be used by the deployment? If you think it's too much for the introductory post, it's good.

Reply:
Good idea, added in https://github.com/kubewarden/kubewarden.io/pull/33/commits/fe3faa37d1105e3cb847c9d79b24911697ec20be.